src/utils.php

```diff
@@ -645,3 +645,87 @@ function ensure_dir( $dir ) {
 
 	return realpath( $dir );
 }
+
+/**
+ * Recursively check permissions in a directory and confirm the UID/GID is properly set.
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return string|bool Error message or true if all checks pass.
+ */
+function check_permissions_in_directory_recursively( string $dir ) {
+	// Open the directory.
+	if ( $handle = opendir( $dir ) ) {
+		// Loop through each entry in the directory.
+		while ( false !== ( $entry = readdir( $handle ) ) ) {
+			// Skip '.' and '..' directories.
+			if ( $entry === '.' || $entry === '..' ) {
+				continue;
+			}
+
+			// Construct the full file path.
+			$file_path = $dir . '/' . $entry;
+
+			// Define the invalid ID and the error message.
+			$invalid_id = 100999;
+			$message    = "Slic has identified that the file located at ${file_path} is assigned a UID/GID of ${invalid_id}. This is likely a result of a documented Docker bug (refer to: https://github.com/docker/desktop-linux/issues/31). To address this issue, execute the following command: `sudo chown -R user_id:group_id ${file_path}`. Make sure to substitute 'user_id' and 'group_id' with the appropriate values.";
+
+			// Check if the file is readable.
+			if ( ! is_readable( $file_path ) ) {
```

Comment:
The file might not be readable because the file modes are wrong; I'm not sure, but it looks like each file should be checked for ownership, not readability.